Hotfix/13248_IllegalArgumentException org.wordpress.android.editor.gutenberg.GutenbergPropsBuilder in build

marecar3 opened the pull request:
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/13248

To test:

At this moment it's not clear the exact scenario of how this crash can be reproduced.
I can assume that maybe as we are retaining `GutenbergEditorFragment` which is creating `GutenbergContainerFragment` it survives rotation while activity is null?

@mchowning @jd-alexander if you have ideas on how to reproduce the crash it would be helpful. Thanks.


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jkmassel committed:
Bump version number

marecar3 committed:
Make sure that getActivity() isn't null before sending capabilities

marecar3 committed:
Make sure that getActivity() isn't null before sending capabilities

marecar3 committed:
Update release notes

marecar3 committed:
Update release notes

mchowning commented on libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergContainerFragment.java:
```diff
@@ -212,7 +212,11 @@ public void updateTheme(Bundle editorTheme) {
+        // We want to make sure that activity isn't null
+        // as it can make this crash to happen: https://github.com/wordpress-mobile/WordPress-Android/issues/13248
+        if (getActivity() != null) {
+            GutenbergProps gutenbergProps = gutenbergPropsBuilder.build(getActivity(), mHtmlModeEnabled);
```
Perhaps extract `getActivity()` out to a variable to just insure that we never hit the scenario where the call on line 217 an object, but the second `getActivity()` call on line 218 to returns `null`.

```
final Activity activity = getActivity();
if (activity !=null) {
  GutenbergProps gutenbergProps = gutenbergPropsBuilder.build(activity, mHtmlModeEnabled);
  ...
}
```

marecar3 replied:
Ah good one. Fixed!

mchowning reviewed (approved):
I also wasn't able to find a way to recreate the crash either. I tested this by doing some smoke-testing both with your changes and with everything inside the if-not-null check commented out (to do a rough simulation of what might happen if the activity was `null`), and I didn't see any issues.

I did leave one small suggested change, but I do not think it is critical.

marecar3 committed:
Extract getActivity() out to a variable

marecar3 committed:
Merge branch 'hotfix/13248_IllegalArgumentException-org.wordpress.android.editor.gutenberg.GutenbergPropsBuilder-in-build' of https://github.com/wordpress-mobile/WordPress-Android into hotfix/13248_IllegalArgumentException-org.wordpress.android.editor.gutenberg.GutenbergPropsBuilder-in-build

# Conflicts:
#	libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergContainerFragment.java

marecar3 committed:
Update release notes

mokagio committed:
Update versionCode in build.gradle to 943

The latest versionCode on the Play Console is 942 from the version named
alpha-252, published on October 26, 2020.

mokagio commented on WordPress/build.gradle:
```diff
@@ -92,9 +92,9 @@ android {
+                versionName "15.9.2"
             }
-            versionCode 938
+            versionCode 943
```
The latest `versionCode` on the Play Console was 942.

The issue that made CI fail [here](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/17778/workflows/dc3da90b-844d-4784-9234-33320a4e5918/jobs/79995) was that `versionCode` was the SemVer style value `15.9.2` instead of an integer.

I'm not sure how this happened in this instance, but I did the same thing once because I didn't read the prompt from the hotfix lane properly and input the `versionName` value when asked for `versionCode`.

Sounds like something we might want to guard against in the release toolkit. I had a quick look at the time and saw that Fastlane allows you to specify whether an input parameter should be an integer. But, when I tried to do it, the validation failed from CLI input. I haven't had the time to look deeper into it, yet.

cc @wordpress-mobile/release-managers.
